GPU/Common/SplineCommon.cpp

```diff
@@ -926,16 +926,19 @@ void DrawEngineCommon::SubmitSpline(const void *control_points, const void *indi
 	patch.patchFacing = patchFacing;
 
 	if (g_Config.bHardwareTessellation && g_Config.bHardwareTransform && !g_Config.bSoftwareRendering) {
-		float *pos = (float*)(decoded + 65536 * 18); // Size 3 float
-		float *tex = pos + count_u * count_v * 3; // Size 3 float
-		float *col = tex + count_u * count_v * 3; // Size 4 float
+		int stride = g_Config.iGPUBackend == GPU_BACKEND_VULKAN || g_Config.iGPUBackend == GPU_BACKEND_DIRECT3D11 ? 4 : 3;
+		float *pos = (float*)(decoded + 65536 * 18); // Size 3 float (4 for Vulkan and D3D11)
+		float *tex = pos + count_u * count_v * stride; // Size 3 float (4 for Vulkan and D3D11)
+		float *col = tex + count_u * count_v * stride; // Size 4 float
 		const bool hasColor = (origVertType & GE_VTYPE_COL_MASK) != 0;
 		const bool hasTexCoords = (origVertType & GE_VTYPE_TC_MASK) != 0;
 
+		if (g_Config.iGPUBackend == GPU_BACKEND_VULKAN)
```

Comment:
We can just call this always, right?  I mean, the other backends, it will do nothing.  This makes it easier to add optimizations to the other backends later without wondering why it doesn't seem to be working.

-[Unknown]